src/codegen/compile.ml

```diff
@@ -4775,6 +4775,20 @@ module Serialization = struct
         get_x
       in
 
+      let read_principal () =
+        let (set_len, get_len) = new_local env "len" in
+        let (set_x, get_x) = new_local env "x" in
+        ReadBuf.read_leb128 env get_data_buf ^^ set_len ^^
+
+        get_len ^^ compile_unboxed_const 23l ^^ G.i (Compare (Wasm.Values.I32 I32Op.LeU)) ^^
```

Comment:
I feel some disconnect here. In #2875 you say that principals are at most 29 bytes, but you set the limit to 23 in this line.

Reply:
Thanks for catching! The number 23 entered my mind somehow between writing the issue and writing the code